jerry-core/ecma/operations/ecma-objects.c

```diff
@@ -663,6 +722,35 @@ ecma_op_object_put (ecma_object_t *object_p, /**< the object */
       }
       break;
     }
+    case ECMA_OBJECT_TYPE_CLASS:
+    {
+#ifndef CONFIG_DISABLE_TYPEDARRAY_BUILTIN
+      /* ES2015 9.4.5.5 */
+      if (ecma_is_typedarray (ecma_make_object_value (object_p)))
+      {
+        ecma_number_t num = ecma_string_to_number (property_name_p);
+        ecma_string_t *num_to_str = ecma_new_ecma_string_from_number (num);
```

Comment:
[9.4.5.5.a](http://www.ecma-international.org/ecma-262/6.0/#sec-integer-indexed-exotic-objects-set-p-v-receiver): the prop name "3.1" can be [converted to number](http://www.ecma-international.org/ecma-262/6.0/#sec-canonicalnumericindexstring) 

9.4.5.5 2.c.i, in [IntegerIndexedElementSet](http://www.ecma-international.org/ecma-262/6.0/#sec-integerindexedelementset) the number 3.1 is not integer, return false

So I think V8's implementation conforms to the spec.